### PR TITLE
Re-adds now-supported clang-format options

### PR DESCRIPTION
**Pull request by cugone**
There is an option in Visual Studio to only apply clang-format changes on Keyboard combo `Ctrl+K, Ctrl+D` to the whole document or `Ctrl+K,Ctrl+F` for the current selection only.

`Tools > Options > Text Editor > C/C++ > Formatting > General > Clang Format execution > check "Run ClangFormat only for manually invoked formatting commands"`

This would have been good to know earlier... Oh well.

Here's a [reference guide](https://clang.llvm.org/docs/ClangFormatStyleOptions.html) to the commands.

There are no anchor links, sorry. :(

You'll have to Ctrl+F search for the command....or just read through the entire document like I did. :/

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -67,21 +64,11 @@ SortIncludes : true
...
-
+SpaceBeforeCpp11BracedList: false
+SpaceBeforeCtorInitializerColon: true
+SpaceBeforeInheritanceColon: true
 SpaceBeforeParens: ControlStatements
-
-#Unsupported in VS2017-included ClangFormat 6.0 we'll see about VS2019's ClangFormat 7.0
-#SpaceBeforeRangeBasedForLoopColon: false
-
+SpaceBeforeRangeBasedForLoopColon: false
```
Comment (DanRStevens) [inline .clang-format]: I suspect we want this to be `true`. Existing uses of range for put a space before the colon.

File: .clang-format [hunk 2]
```diff
@@ -67,21 +64,11 @@ SortIncludes : true
...
-#Unsupported in VS2017-included ClangFormat 6.0 we'll see about VS2019's ClangFormat 7.0
-#SpaceBeforeCpp11BracedList: false
-
-#Unsupported in VS2017-included ClangFormat 6.0 we'll see about VS2019's ClangFormat 7.0
-#SpaceBeforeCtorInitializerColon: true
-
-#Unsupported in VS2017-included ClangFormat 6.0 we'll see about VS2019's ClangFormat 7.0
-#SpaceBeforeInheritanceColon: true
-
+SpaceBeforeCpp11BracedList: false
```
Comment (DanRStevens) [inline .clang-format]: Would be nice to know what exactly this means. Maybe a reference link would help.

In particular, does this setting describe existing practice?

File: .clang-format [hunk 3]
```diff
@@ -56,8 +54,7 @@ IncludeCategories:
     - Priority: 1
 
 IndentCaseLabels: false
-#Unsupported in VS-included ClangFormat 6.0 we'll see about VS2019's ClangFormat 7.0
-#IndentPPDirectives: BeforeHash
+IndentPPDirectives: BeforeHash
```
Comment (DanRStevens) [inline .clang-format]: I have no idea what this one does.

File: .clang-format (hunk 2 above)
Comment (cugone) [reply .clang-format]: Ref: https://clang.llvm.org/docs/ClangFormatStyleOptions.html
There are no anchor links, sorry. :(
You'll have to Ctrl+F search for the command.

> If true, a space will be inserted before a C++11 braced list used to initialize an object (after the preceding identifier or type).
```
    true:                                  false:
    Foo foo { bar };               vs.     Foo foo{ bar };
    Foo {};                                Foo{};
    vector<int> { 1, 2, 3 };               vector<int>{ 1, 2, 3 };
    new int[3] { 1, 2, 3 };                new int[3]{ 1, 2, 3 };
```

Comment (cugone) [reply .clang-format]: The false option does.

File: .clang-format (hunk 3 above)
Comment (cugone) [reply .clang-format]: Ref: https://clang.llvm.org/docs/ClangFormatStyleOptions.html
>IndentPPDirectives (PPDirectiveIndentStyle)
>
>    The preprocessor directive indenting style to use.
>
> Possible values:
>
- `PPDIS_None` (in configuration: `None`) Does not indent any directives.
```
        #if FOO
        #if BAR
        #include <foo>
        #endif
        #endif
```
- `PPDIS_AfterHash` (in configuration: `AfterHash`) Indents directives after the hash.
```
        #if FOO
        #  if BAR
        #    include <foo>
        #  endif
        #endif
```
- `PPDIS_BeforeHash` (in configuration: `BeforeHash`) Indents directives before the hash.
```
        #if FOO
          #if BAR
            #include <foo>
          #endif
        #endif
```

File: .clang-format (hunk 1 above)
Comment (cugone) [reply .clang-format]: Point. Oversight on my part.

**Pull request merged**